hmt_escrow/storage.py

```diff
@@ -70,9 +70,49 @@ def download(key: str, private_key: bytes) -> Dict:
     msg = _decrypt(private_key, ciphertext)
     return json.loads(msg)
 
+def createNewIpnsLink(name: str) -> str:
+    """Creates a new IPFS Id. The IPNS links are managed by key value system.
+       Pass in the key (e.g. 'intermediate-results-' + self.job_contract.address), 
+       then creats the ipns link
 
+    >>> import random 
+    >>> keyName = str(random.getrandbits(32 * 8)) # get random, or else throws duplicate key error
+    >>> createNewIpnsLink(keyName) != ''
+    True
+
+    Args:
+        name (str): Name to call the ipns link. Retrieve the link w/ same name
+
+    Returns:
+        str: Returns the IPNS url
+    """
+    name = name.lower()
+    key = IPFS_CLIENT.key.gen(name, 'ed25519')
+    return getIpnsLink(name)
+
+def getIpnsLink(name: str) -> str:
+    """Get the ipns link with the name of it which we remember it by
+
+    >>> import random 
+    >>> keyName = str(random.getrandbits(32 * 8)) # get random, or else throws duplicate key error
+    >>> createNewIpnsLink(keyName) != ''
+    True
+
+    Args:
+        name (str): Name we call ipns link
+
+    Returns:
+        str: Returns the IPNS url
+    """
+    keys = IPFS_CLIENT.key.list()
+    try:
+        return 'https://ipfs.io/ipns/' + list(filter(lambda x: x['Name'] == name.lower(), keys['Keys']))[0]['Id']
+    except Exception as e:
+        return ''
+    return ''
```

Comment:
why not let the exceptions occur here? if theres a problem you don't want to have to write additional code to check for a blank string

Reply:
The thought was that if there is no key in the list, then you would not want to throw an exception. Maybe we can do a test beforehand? `nameExists(name) -> bool`

Reply:
Hm, I see.. maybe `None` is the better response vs an empty string?  Idk my preference here would be to more narrowly do a try/catch and throw an exception if there is something they're doing wrong from the code perspective (ie if no name found, then `raise ValueError('no name found')`. 

If the code and usage of the api is correct, and a None/empty string is a valid response that can be used just like a real url, then empty string might be right.

Reply:
IMO it's better to be explicit about why something fails, and throw a specific exception so that higher up we can differentiate where the problem is - ie is it a user input failure or is it a code failure that we might want to handle different ways